test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -76,6 +76,8 @@ private void assertCommandBehavior(String inputCommand,
                                       AddressBook expectedAddressBook,
                                       boolean isRelevantPersonsExpected,
                                       List<? extends ReadOnlyPerson> lastShownList) throws Exception {
+        //Save the state
+        saveFile.save(addressBook);
```

Comment:
This makes the test pass, but does the test really check for the correct command behaviour now?
Does your test check that a non-mutating command doesn't save the file, and vice versa?